Code/RDGeneral/Invariant.cpp

```diff
@@ -46,4 +47,35 @@ std::string Invariant::toString() const {
 
   return stringRep;
 }
+
+std::string Invariant::toUserString() const {
+  std::string line = boost::lexical_cast<std::string>(this->getLine());
+
+  std::string filename = this->getFile();
+  
+  std::size_t pos = filename.find("Code"); // strip out build directory info
+  if (pos != std::string::npos) {
+    filename = filename.substr(pos);
+  }
+  
+  std::string stringRep = this->prefix_d + "\n\t" + this->getMessage() +
+      "\n\tViolation occurred on line " + line + " in file " +
+      filename + "\n\tFailed Expression: " +
+      this->getExpression() + "\n\t" +
+      "RDKIT: " + RDKit::rdkitVersion + "\n\t" +
```

Comment:
I could just put this in toString and have just one function.  I kind of like having the full path for dev purposes which is perhaps a bit selfish :)